src/psyclone/domain/nemo/transformations/nemo_outerarrayrange2loop_trans.py

```diff
@@ -0,0 +1,186 @@
+        UBOUND intrinsics are used as appropriate. The type of the
+        NemoLoop is also taken from the configuration file if it is
+        supplied for that index, otherwise it is specified as being
+        "unknown".
+
+        :param node: an Assignment node.
+        :type node: :py:class:`psyclone.psyir.nodes.Assignment`
+        :param options: a dictionary with options for \
+            transformations. No options are used in this \
+            transformation. This is an optional argument that defaults \
+            to None.
+        :type options: dict of string:values or None
+
+        '''
+        self.validate(node)
+
+        # get lhs array
+        lhs_array_ref = node.lhs
+        index = _get_outer_index(lhs_array_ref)
+        nemo_arrayrange2loop = NemoArrayRange2LoopTrans()
+        nemo_arrayrange2loop.apply(lhs_array_ref.children[index])
+
+    def __str__(self):
+        return ("Convert a PSyIR assignment to the outermost ArrayReference "
+                "Range into a PSyIR NemoLoop.")
+
+    @property
+    def name(self):
+        '''
+        :returns: the name of the transformation.
+        :rtype: str
+
+        '''
+        return type(self).__name__
+
+    def validate(self, node, options=None):
+        '''Perform various checks to ensure that it is valid to apply the
+        NemoArrayRange2LoopTrans transformation to the supplied PSyIR Node.
+
+        :param node: the node that is being checked.
+        :type node: :py:class:`psyclone.psyir.nodes.Assignment`
+        :param options: a dictionary with options for \
+            transformations. No options are used in this \
+            transformation. This is an optional argument that defaults \
+            to None.
+        :type options: dict of string:values or None
+
+        :raises TransformationError: if the supplied node is an \
+            Assignment node, if the Assignment node does not have an \
+            ArrayReference node on its left hand side or if the \
+            ArrayReference node does not contain at least one Range \
+            node.
+
+        '''
+        # Am I an assignment node?
+        if not isinstance(node, Assignment):
+            raise TransformationError(
+                "Error in NemoOuterArrayRange2LoopTrans transformation. The "
+                "supplied node argument should be a PSyIR Assignment, but "
+                "found '{0}'.".format(type(node).__name__))
+
+        # Is the LHS an array reference?
+        if not isinstance(node.lhs, ArrayReference):
+            raise TransformationError(
+                "Error in NemoOuterArrayRange2LoopTrans transformation. The "
+                "supplied assignment node should have an ArrayReference node "
+                "on its lhs but found '{0}'."
+                "".format(type(node.parent).__name__))
+        array_reference = node.lhs
+        # Has the array reference got a range?
+        try:
+            _ = _get_outer_index(array_reference)
```

Comment:
Since we're using this method in this module, it can no longer claim to be private and therefore shouldn't have the leading underscore in its name.

Reply:
This is a situation where I don't want the function visible to the user via the API but I do want to use it in more than one class (in different files). There does not seem to be a Python solution for this and I went for using `_` to hide it from the API. However, I'm happy to remove it as requested.